hugo/content/tutorials/langium_and_monaco.md

```diff
@@ -169,233 +214,368 @@ h1 {
-});
+Lastly, let's setup the user config, which will be used to startup the wrapper.
 
-// setup the language client config with the worker
-const languageClientConfig = {
-    enabled: true,
-    useWebSocket: false,
-    // can pass configuration data, or a pre-configured worker as well
-    // the later works better for us in this case
-    workerConfigOptions: lsWorker
-};
-```
+```ts
+type WorkerUrl = string;
 
-We can also pass more explicit config options to `workerConfigOptions` if we don't want to create the worker ourselves. For example:
+/**
+ * Generalized configuration used with 'getMonacoEditorReactConfig' to generate a working configuration for monaco-editor-react
+ */
+interface ClassicConfig {
+    code: string,
+    htmlElement: HTMLElement,
+    languageId: string,
+    worker: WorkerUrl | Worker,
+    monarchGrammar: any;
+}
 
-```js
-{
-    url: new URL('./minilogo-server-worker.js', import.meta.url),
-    type: 'classic',
-    name: 'minilogo-language-server-worker'
+/**
+ * Generates a UserConfig for a given Langium example, which is then passed to the monaco-editor-react component
+ * 
+ * @param config A VSCode API or classic editor config to generate a UserConfig from
+ * @returns A completed UserConfig
+ */
+function createUserConfig(config: ClassicConfig): UserConfig {
+    // setup urls for config & grammar
+    const id = config.languageId;
+
+    // generate langium config
+    return {
+        htmlElement: config.htmlElement,
+        wrapperConfig: {
+            editorAppConfig: {
+                $type: 'classic',
+                languageId: id,
+                useDiffEditor: false,
+                automaticLayout: true,
+                code: config.code,
+                theme: 'vs-dark',
+                languageDef: config.monarchGrammar
+            },
+            serviceConfig: {
+                enableModelService: true,
+                configureConfigurationService: {
+                    defaultWorkspaceUri: '/tmp/'
+                },
+                enableKeybindingsService: true,
+                enableLanguagesService: true,
+                debugLogging: false
+            }
+        },
+        languageClientConfig: {
+            options: {
+                $type: 'WorkerDirect',
+                worker: config.worker as Worker,
+                name: `${id}-language-server-worker`
+            }
+        }
+    };
 }
 ```
 
-Either case works, but by creating the worker in advance, we give ourselves the ability to directly interact with the worker/LS independent of the wrapper itself, and to even pre-configure it before use. This can hugely beneficial, especially if we expect to customize our LS on the fly.
+This particular UserConfig will be for configuring a classic editor, rather than a VSCode API editor. This is because we're using a Monarch grammar, which is not supported by the VSCode API. However, if we wanted to use a TextMate grammar, we could use the VSCode API type instead.
```

Comment:
What about:
This particular UserConfig will be for configuring a classic editor, rather than a VSCode extension based editor. This is because we're using a Monarch grammar, which is not supported by the later. However, if we wanted to use a TextMate grammar, we could should use the extension based configuration instead.

Reply:
You can copy text from the updated README here:
https://github.com/TypeFox/monaco-components/pull/54

The text/comment above was one of the drivers to make this clearer.

Reply:
Good suggestion, I'll add this in.